lusidtools/pandas_utils/lusid_pandas.py

```diff
@@ -0,0 +1,41 @@
+import pandas as pd
+from flatten_json import flatten
+
+
+def lusid_respose_to_pd(lusid_response):
+
+    """
+    This function takes a LUSID API response and attempts to convert the response into a Pandas DataFrame or Series.
+    Not all LUSID API responses are the same datatype.
+    Therefore we need to implement some if/else conditional logic to see how the response should be converted.
+
+    :param lusid_response: a response from the LUSID APIs (e.g. VersionedResourceListOfPortfolioHolding)
+
+    """
+
+    if hasattr(lusid_response, "values") and type(lusid_response.values) == list:
+
+        # The API responses which return a list of LUSID objects have a "values" attribute
+        # These "values" attributes are Python lists
+        # Examples include "Get Holdings" where a list of holding objects is returned
+        # Another example is "List Instruments" where a list of instrument objects is returned
+
+        pandas_object = pd.DataFrame(
+            flatten(value.to_dict(), ".") for value in lusid_response.values
+        )
+
+    elif hasattr(lusid_response, "to_dict"):
+
+        # The API responses for single LUSID objects have a "to_dict" attribute
+        # Example "Get Instrument" returns a dictionary with some key-value pairs for one instrument
+
+        pandas_object = pd.Series(flatten(lusid_response.to_dict(), "."))
+
+    else:
+
+        raise TypeError(
```

Comment:
I wonder if rather than throwing an exception we convert the whole response raw? If you are using `lusidtools` for general use (i.e. not necessarily using `load_from_data_frame`) you might want to get the response and pre-process in a DataFrame in which case this would be a useful function 

@MikeMcGarry @thomasdotred what do you think?

Reply:
What do you mean by convert raw? The only exception here is if there is no to_dict method right? I think every model should have this method.

Also @slemasne I think this should always return a Pandas DataFrame. The Series and DataFrame have different attributes and methods (interfaces) so with this function you will never be sure what you will get back.

Reply:
sorry I meant to convert any arbitrary object to a DataFrame

Reply:
Thanks. Thats updated.